### DISCUSSION TITLE
Changing args on a useContractWrite

### DISCUSSION THREAD

**Discussion by hellowodl** [unicape/use-wagmi#81] [Q&A] [answered]
Hey there :)

I'm currently facing an issue where if I use a useContractWrite, the args will always be the initial args, and I can't find any hints in the API regarding how to refresh that.

Any ideas?

**Comment by unicape** [comment]
@hellowodl You need to check if the parameter is reactive

**Reply by hellowodl** [reply]
As in provide a vue reactive()?

I've solved it by passing args to write(), though that didn't resolve the simulation.

**Reply by hellowodl** [reply]
I was basically doing `{ args: [computedValue.value] }`

**Reply by unicape** [reply]
```js
const address = ref('0x00...')
const arg1 = ref('123')
useContractWrite({
  abi,
  address,
  functionName: 'xxx',
  args: [arg1]
})
```

**Reply by unicape** [reply, marked as answer]
`{ args: [computedValue] }` This is correct

**Reply by hellowodl** [reply]
Oh crap. I didn't even know that was possible. Awesome, thanks :)
